internal/output/tls/tls.go

```diff
@@ -78,5 +78,13 @@ func (o *Output) Close() error {
 
 // Write writes data to the TLS connection.
 func (o *Output) Write(b []byte) (int, error) {
-	return o.conn.Write(append(b, '\n'))
+	if o.conn == nil {
+		return 0, errors.New("not connected")
+	}
+
+	// Add a newline for framing.
+	buf := make([]byte, len(b)+1)
+	copy(buf, b)
+	buf[len(b)] = '\n'
+	return o.conn.Write(buf)
```

Comment:
In both cases this could be `return o.conn.Write(append(b[:len(b):len(b)], '\n'))` with safety.